honeybee_radiance/lightsource/sky/sunmatrix.py

```diff
@@ -0,0 +1,192 @@
+            the North and the positive Y-axis in degrees. 90 is West and 270 is East
+            (Default: 0)
+
+    Properties:
+        * wea
+        * location
+        * north
+        * is_point_in_time
+        * is_climate_based
+    """
+
+    __slots__ = ('_wea', '_north')
+
+    def __init__(self, wea, north=0):
+        """Create a climate-based sun matrix."""
+        self.wea = wea
+        self.north = north
+
+    @property
+    def wea(self):
+        """Get and set wea."""
+        return self._wea
+
+    @wea.setter
+    def wea(self, value):
+        assert isinstance(value, Wea), \
+            'wea must be from type Wea not {}'.format(type(value))
+        self._wea = value
+
+    @property
+    def north(self):
+        """Get and set north direction.
+
+        A number between -360 and 360 for the counterclockwise difference between
+        the North and the positive Y-axis in degrees. 90 is West and 270 is East.
+        """
+        return self._north
+
+    @north.setter
+    def north(self, value):
+        value = typing.float_in_range(value, -360, +360, 'Skymatrix north')
+        self._north = value
+
+    @property
+    def is_climate_based(self):
+        """Return True if the sky is created based on values from weather data."""
+        return True
+
+    @property
+    def is_point_in_time(self):
+        """Return True if the sky is generated for a single point in time."""
+        return False
+
+    @property
+    def location(self):
+        """Location information for sky matrix."""
+        return self.wea.location
+
+    @classmethod
+    def from_dict(cls, input_dict):
+        """Create the sky from a dictionary.
+
+        Args:
+            input_dict: A python dictionary in the following format
+
+        .. code-block:: python
+
+            {
+                'type': 'SunMatrix',
+                'wea': {},
+                'north': 0.0  # optional
+            }
+        """
+        if 'type' not in input_dict or input_dict['type'] != 'SunMatrix':
+            raise ValueError('Input dict "type" must be "SunMatrix".')
+        if 'north' in input_dict:
+            sky = cls(Wea.from_dict(input_dict['wea']), input_dict['north'])
+        else:
+            sky = cls(input_dict['wea'])
+
+        return sky
+
+    # TODO: add support for additional parameters
+    # TODO: add gendaymtx to radiance-command and use it for validating inputs
+    def to_radiance(self, output_type=1, wea_file=None, output_name=None):
+        """Return Radiance command to generate the sky.
+
+        Note that you need to write the wea to a file (in.wea) before running this
+        command.
+
+        Alternatively you can use write method which will write the wea data to a file.
+
+        Args:
+            output_type: An integer between 0 to 2 for output type.
+                * 0 = output in W/m2/sr visible
+                * 1 = output in W/m2/sr solar (default)
```

Comment:
If this is `an integer between 0 and 2`, then what does `2` signify? 

If 2 doesn't mean anything and this is only ever two options (visible and solar), maybe a boolean input is more appropriate.

Reply:
Boolean is an option but using an integer for output_type will keep it consistent among different skies.